From 40a13a610884ad176e9197eefe4a8bef09f94c19 Mon Sep 17 00:00:00 2001
From: okuoku <mjt@cltn.org>
Date: Sat, 20 Mar 2021 14:16:47 +0900
Subject: [PATCH] Build: Allow bootstrap directly

---
 Makefile | 25 ++++++++++++++++++-------
 1 file changed, 18 insertions(+), 7 deletions(-)

diff --git a/Makefile b/Makefile
index d442b97b..0bad3ad8 100644
--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,14 @@
 include Makefile.config
 
 # Commands
-CYCLONE = cyclone -A .
+ROOT = $(CURDIR)
+CYC_CP = "$(CC) ~src-file~ -Iinclude $(COMP_PROG_CFLAGS) -c -o ~exec-file~.o"
+CYC_CE = "$(CC) ~exec-file~.o ~obj-files~ -L. $(LIBS) $(COMP_CFLAGS) -o ~exec-file~"
+CYC_CL = "$(CC) ~src-file~ -Iinclude $(COMP_CFLAGS) -c -o ~exec-file~.o"
+CYC_CS = "$(CC) -shared -L. $(LDFLAGS) -o ~exec-file~.so ~exec-file~.o"
+CYCLONE = cyclone -no-batch -I $(ROOT) -I $(ROOT)/libs -CP $(CYC_CP) -CE $(CYC_CE) -CL $(CYC_CL) -CS $(CYC_CS)
+CYCLONE0 = $(ROOT)/cyclone0 -no-batch -I $(ROOT) -I $(ROOT)/libs -CP $(CYC_CP) -CE $(CYC_CE) -CL $(CYC_CL) -CS $(CYC_CS)
+CYCLONE_BOOT = cyclone -no-batch -I $(ROOT)
 CCOMP = $(CC) $(CFLAGS)
 INDENT_CMD = indent -linux -l80 -i2 -nut
 
@@ -158,14 +165,18 @@ hello-library/hello :
 
 libs : $(COBJECTS)
 
-$(COBJECTS) : %.o: %.sld
-	$(CYCLONE) $<
+$(COBJECTS) : %.o: %.sld cyclone0
+	$(CYCLONE0) $<
 
-cyclone : cyclone.scm $(CYC_RT_LIB) $(CYC_BN_LIB)
-	$(CYCLONE) cyclone.scm
+cyclone0 : cyclone.scm $(CYC_RT_LIB) $(CYC_BN_LIB)
+	$(CYCLONE) cyclone.scm && mv ./cyclone ./cyclone0
+
+cyclone : cyclone.scm $(CYC_RT_LIB) $(CYC_BN_LIB) cyclone0 libs
+	$(CYCLONE0) cyclone.scm
+
+icyc : icyc.scm $(CYC_RT_LIB) $(CYC_BN_LIB) libs
+	$(CYCLONE0) $<
 
-icyc : icyc.scm $(CYC_RT_LIB) $(CYC_BN_LIB)
-	$(CYCLONE) $<
 
 $(CYC_RT_LIB) : $(CFILES) $(HEADERS) $(CYC_BN_LIB)
 
-- 
2.25.1

